--- src/core/Connector.h
+++ src/core/Connector.h
@@ -2,7 +2,9 @@
 #define CONNECTOR_H

 #include "Item.h"
+#include "HttpApiEndpoint.h"
 #include <QString>
+#include <QList>

 class Connector {
 public:
@@ -17,6 +19,16 @@
     return nullptr;
   }
   virtual void saveSettings(class QWidget *settingsWidget) {}
+
+  virtual bool hasDebugMenu() const { return false; }
+  virtual QList<HttpApiEndpoint> getHttpApiEndpoints() const {
+    return QList<HttpApiEndpoint>();
+  }
+  virtual QMap<QString, QString> getApiSubstitutions() const {
+    return QMap<QString, QString>();
+  }
 };

 #define Connector_iid "com.kmagmux.Connector/1.0"
